### ISSUE TITLE
[Bug?] Maybe be there are some mistake about use of @Qualifer 

### ISSUE THREAD

**Issue by csh0101** [FeatureProbe/FeatureProbe#297]
I think the @Qualifer in the UserPasswordAuthenicateProvider is not useful. 
``` yaml
    validator:
      impl: common
```
I just add the code following this:
``` java
@Component
@AllArgsConstructor
@Qualifier("test")
public class TestAccountValidator implements AccountValidator{

    @Override
    public Authentication authenticate(Authentication authentication) {
        return null;
    }
}
```
and run application , but error 
- commonAccountValidator: defined in file [/Users/csh0101/lab/java-playground/FeatureProbe/api/admin/target/classes/io/featureprobe/api/auth/CommonAccountValidator.class]
	- testAccountValidator: defined in file [/Users/csh0101/lab/java-playground/FeatureProbe/api/admin/target/classes/io/featureprobe/api/auth/TestAccountValidator.class]
	
so I have the choice to think about the @Qualifer is not useful. Maybe the use is mistake 
this some reason I think:
@AllArgsConstruct will cause the spring ignore the @Qualifer, so you maybe need refix code like this :
    @Autowired
    @Qualifier("test")
    AccountValidator validator;  
anything else, I think the spring el to get value from config file is not useful in @Qualifier 
see this err:
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'io.featureprobe.api.auth.AccountValidator' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.springframework.beans.factory.annotation.Qualifier(value=${app.security.validator.impl})}

## consulition
  you ignore this mistake ,just because there are only one bean that named CommonValidator. even the @Qualifier is useless, but the DI still work. but in now, the TestValidator class is  added, and the mistake is appear.
 
 ## WorkAround
  Because of I was develop the feat of ldap, so I will do @Priamry work around this problem until we fix it.

**Comment by csh0101** [comment]
@FengShang

**Comment by csh0101** [comment]
And Maybe the validator can be a List? because in method support,it look likes strategy design pattern

**Comment by gangb-tech** [comment]
> I think the @Qualifer in the UserPasswordAuthenicateProvider is not useful.
> 
> ```yaml
>     validator:
>       impl: common
> ```
> 
> I just add the code following this:
> 
> ```java
> @Component
> @AllArgsConstructor
> @Qualifier("test")
> public class TestAccountValidator implements AccountValidator{
> 
>     @Override
>     public Authentication authenticate(Authentication authentication) {
>         return null;
>     }
> }
> ```
> 
> and run application , but error
> 
> * commonAccountValidator: defined in file [/Users/csh0101/lab/java-playground/FeatureProbe/api/admin/target/classes/io/featureprobe/api/auth/CommonAccountValidator.class]
>   
>   * testAccountValidator: defined in file [/Users/csh0101/lab/java-playground/FeatureProbe/api/admin/target/classes/io/featureprobe/api/auth/TestAccountValidator.class]
> 
> so I have the choice to think about the @Qualifer is not useful. Maybe the use is mistake this some reason I think: @AllArgsConstruct will cause the spring ignore the @Qualifer, so you maybe need refix code like this : @Autowired @qualifier("test") AccountValidator validator; anything else, I think the spring el to get value from config file is not useful in @qualifier see this err: Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'io.featureprobe.api.auth.AccountValidator' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.springframework.beans.factory.annotation.Qualifier(value=${app.security.validator.impl})}
> 
> ## consulition
> you ignore this mistake ,just because there are only one bean that named CommonValidator. even the @qualifier is useless, but the DI still work. but in now, the TestValidator class is added, and the mistake is appear.
> 
> ## WorkAround
> Because of I was develop the feat of ldap, so I will do @Priamry work around this problem until we fix it.

Yes, you are right. We are currently refactoring the code, and it is not yet a stable version. This issue has already been fixed in the next iteration.

**Comment by gangb-tech** [comment]
> > I think the @Qualifer in the UserPasswordAuthenicateProvider is not useful.
> > ```yaml
> >     validator:
> >       impl: common
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I just add the code following this:
> > ```java
> > @Component
> > @AllArgsConstructor
> > @Qualifier("test")
> > public class TestAccountValidator implements AccountValidator{
> > 
> >     @Override
> >     public Authentication authenticate(Authentication authentication) {
> >         return null;
> >     }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and run application , but error
> > 
> > * commonAccountValidator: defined in file [/Users/csh0101/lab/java-playground/FeatureProbe/api/admin/target/classes/io/featureprobe/api/auth/CommonAccountValidator.class]
> >   
> >   * testAccountValidator: defined in file [/Users/csh0101/lab/java-playground/FeatureProbe/api/admin/target/classes/io/featureprobe/api/auth/TestAccountValidator.class]
> > 
> > so I have the choice to think about the @Qualifer is not useful. Maybe the use is mistake this some reason I think: @AllArgsConstruct will cause the spring ignore the @Qualifer, so you maybe need refix code like this : @Autowired @qualifier("test") AccountValidator validator; anything else, I think the spring el to get value from config file is not useful in @qualifier see this err: Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'io.featureprobe.api.auth.AccountValidator' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true), @org.springframework.beans.factory.annotation.Qualifier(value=${app.security.validator.impl})}
> > ## consulition
> > you ignore this mistake ,just because there are only one bean that named CommonValidator. even the @qualifier is useless, but the DI still work. but in now, the TestValidator class is added, and the mistake is appear.
> > ## WorkAround
> > Because of I was develop the feat of ldap, so I will do @Priamry work around this problem until we fix it.
> 
> Yes, you are right. We are currently refactoring the code, and it is not yet a stable version. This issue has already been fixed in the next iteration.

The issue has been fixed in version 2.6.0. Please pull the code again to get the latest changes.
